src/activities/services/activities.service.ts

```diff
@@ -63,6 +64,46 @@ export class ActivitiesService {
     }
   }
 
+  async updateActivityWithRoutes(
+    activityIn: UpdateActivityInput,
+    user: User,
+    routesIn: CreateActivityRouteInput[],
+  ): Promise<Activity> {
+    const queryRunner = this.connection.createQueryRunner();
+    await queryRunner.connect();
+    await queryRunner.startTransaction();
+
+    try {
+      // TODO: refactor as it breaks DRY heavily, make sure that date did not change, update AR dates from activity date
+      // Create new activity
+      const activity = await this.activitiesRepository.findOneOrFail(
+        activityIn.id,
+      );
```

Comment:
Might change this to:
`const activity = await queryRunner.manager.findOne(Activity, activityIn.id);`
so that query is done within the open transaction.